reactor-netty-incubator-quic/src/test/java/reactor/netty/incubator/quic/QuicServerSmokeTests.java

```diff
@@ -0,0 +1,142 @@
+		doTestServerOpensStream(Flux.just(METHOD, FLUX_PATH, "\r\n"), FLUX_RESPONSE);
+	}
+
+	@Test
+	void testMono() throws Exception {
+		doTestServerOpensStream(Mono.just(METHOD + MONO_PATH + "\r\n"), MONO_RESPONSE);
+	}
+
+	@Test
+	void testMonoError() throws Exception {
+		doTestServerOpensStream(Mono.just(METHOD + ERROR_PATH_2 + "\r\n"), EMPTY_RESPONSE);
+	}
+
+	@Test
+	void testStreamId() throws Exception {
+		doTestServerOpensStream(Mono.just(METHOD + STREAM_ID + "\r\n"), STREAM_ID_RESPONSE);
+	}
+
+	@Test
+	void testThrowsException() throws Exception {
+		doTestServerOpensStream(Mono.just(METHOD + ERROR_PATH_1 + "\r\n"), EMPTY_RESPONSE);
+	}
+
+	private void doTestServerOpensStream(Publisher<String> body, String expectation)
+			throws Exception {
+		CountDownLatch latch = new CountDownLatch(1);
+		AtomicReference<String> response = new AtomicReference<>("");
+		server =
+				createServer()
+				        .doOnConnection(quicConn ->
+				            quicConn.createStream(QuicStreamType.BIDIRECTIONAL, (in, out) -> {
+				                        in.receive()
+				                          .asString()
+				                          .defaultIfEmpty(EMPTY_RESPONSE)
+				                          .doOnNext(s -> {
+				                              response.set(s);
+				                              latch.countDown();
+				                          })
+				                          .subscribe();
+
+				                          return out.sendString(body);
+				                    })
+				                    .subscribe())
+				        .streamObserve((conn, state) -> {
+				            if (state == CONNECTED) {
+				                conn.addHandlerLast(new LineBasedFrameDecoder(1024));
+				            }
+				        })
+				        .bindNow();
+
+		client =
+				createClient(server::address)
+				        .streamObserve((conn, state) -> {
+				            if (state == CONNECTED) {
+				                conn.addHandlerLast(new LineBasedFrameDecoder(1024));
+				            }
+				        })
+				        .handleStream((in, out) ->
+				            out.sendString(in.receive()
+				                             .asString()
+				                             .flatMap(s -> {
+				                                 if ((METHOD + MONO_PATH).equals(s)) {
+				                                     return Mono.just(MONO_RESPONSE + "\r\n");
+				                                 }
+				                                 else if ((METHOD + FLUX_PATH).equals(s)) {
+				                                     return Flux.just("Hello", " ", "Flux", "!", "\r\n");
+				                                 }
+				                                 else if ((METHOD + ERROR_PATH_1).equals(s)) {
+				                                     throw new RuntimeException("error1");
+				                                 }
+				                                 else if ((METHOD + ERROR_PATH_2).equals(s)) {
+				                                     return Mono.error(new RuntimeException("error2"));
```

Comment:
same question as in the client smoke test, how do these `onError` turn into `EMPTY_RESPONSE` on the other side? (maybe I'm basing my expectation too much on RSocket)